dali/pipeline/operators/reader/loader/loader.h

```diff
@@ -167,14 +168,19 @@ class Loader {
   virtual Index Size() = 0;
 
  protected:
+  virtual void MoveToNextShard(Index current_index) {
+    if (IsNextShard(current_index)) {
+      Reset(!stick_to_shard_);
+    }
+  }
   // Reset reader to the first sample
-  virtual void Reset() = 0;
+  virtual void Reset(bool move_to_beginning) = 0;
 
   // Check if given reader moved to the next shard
   virtual inline bool IsNextShard(Index current_index) {
```

Comment:
This is just a naming issue, but now it checks if we are next shard or if we should just wrap around.

So maybe, something like:
```suggestion
  virtual inline bool ShouldWrapAround(Index current_index) {
```

Reply:
Looking into your rationally ShouldWrapAround is equally wrong to IsNextShard.